### PR TITLE
fix(prow/plugins/approve): approving with full config OWNERS controlled

### PR DESCRIPTION
**Pull request by wuhuizuo**
## What

Fixed: #7690

## How

- Make it fixed without massive refactoring, also do not change the behavior and mostly approve notifies.

### CODE REVIEW

**Review by matthyx — APPROVED** [review]
@wuhuizuo clever... I see what you mean by not refactoring
I did try a refactoring with generics for this plugin https://github.com/kubernetes/test-infra/pull/12440 maybe we could revisit it together?

**Comment by wuhuizuo** [comment]
May the integration test job failed by flaky tests.

**Comment by wuhuizuo** [comment]
> @wuhuizuo clever... I see what you mean by not refactoring I did try a refactoring with generics for this plugin #12440 maybe we could revisit it together?

My pleasure! Looking forward to hearing your specific thoughts.

Also I added a new commit to solve the issue when there is no default filter and the inherit from parent is disabled, PTAL:
https://github.com/kubernetes/test-infra/pull/29970/commits/dcd9b3615ff0f9e551895ba63f091f629caefe28

**Comment by wuhuizuo** [comment]
ping @matthyx @cjwagner 😄 could you review it?

File: prow/plugins/approve/approvers/owners.go [hunk 1]
```diff
@@ -122,16 +145,19 @@ func (o Owners) GetReverseMap(approvers map[string]sets.Set[string]) map[string]
...
 	for _, approver := range allApprovers {
 		filesCanApprove := reverseMap[approver]
-		if filesCanApprove.Intersection(unapproved).Len() > maxCovered {
+		if filesCanApprove.Intersection(unapproved).Len() > maxCovered && !coveredApproversSet.Has(approver) {
 			maxCovered = len(filesCanApprove)
 			bestPerson = approver
 		}
 	}
+
+	// todo: make it better.
```
Comment (matthyx) [inline prow/plugins/approve/approvers/owners.go:159]: 😅

**Review by matthyx — APPROVED** [review]
/lgtm

**Pull request merged**